runtime/src/bank.rs

```diff
@@ -4871,6 +4896,66 @@ impl Bank {
         results
     }
 
+    // Note: this function is not yet used; next PR will call it behind a feature gate
+    #[allow(dead_code)]
+    fn filter_program_errors_and_collect_fee_details(
```

Comment:
sorry for asking this so late in review process. We have a near identical function for the non-detailed version.

Is there a reason we cannot have most of this logic in a separate function (filtering errors) then branch only at fee collection using either details or sum?

Reply:
Good point, this function does not co-exist with original `filter_program_errors_and_collect_fee()`. It will replace it when feature gate is activated. Then the original function will be deleted as part of feature gate clean up. I opt to keep duplicated code to make "remove old code" easier.

Reply:
I still have concerns about the duplicate code, it seems quite sensitive and it seems we should be able to make them a shared function.

The previous function does some additional error checks
```
let lamports_per_signature =
    lamports_per_signature.ok_or(TransactionError::BlockhashNotFound)?;
```
I guess my concern comes from these differences; are we sure nothing relies on this checking that?

Could we:
1. accumulate all 3 variables - (total, base, priority)
2. only credit/burn based on feature-activation at the end of the block

Reply:
Talked offline. It seems fine to duplicate the functions, the additional checks never fail and regardless it will be a clean switch-over to new function under the feature-gate.

Reply:
how about 0e450855d86711a588ec8fb98c04bb7448ec629b to reduce dup a bit -- this part of code is identical before/after, not related to logic this PR changes.